content/en/blog/2023-02-08-flux-pulumi-superpowers.md

```diff
@@ -0,0 +1,161 @@
+---
+author: Michael Bridgen
+date: 2023-02-08 8:30:00+00:00
+title: How Flux and Pulumi give each other superpowers
+description: The Pulumi Kubernetes operator extends Flux to all your infrastructure, and Flux makes the Pulumi operator secure and super easy to work with.
+url: /blog/2023/02/flux-pulumi-superpowers/
+tags: [ecosystem]
+---
+
+# How Flux and Pulumi work together
+
+[Pulumi](https://pulumi.com/) is an "Infrastructure as Code" tool that lets you specify your
+infrastructure as programs written in JavaScript, Python, Java, Go, .NET languages, or YAML. The
+[Pulumi Kubernetes operator](https://github.com/pulumi/pulumi-kubernetes-operator) drives Pulumi
+from Kubernetes, so you can maintain your infrastructure by pushing commits to git and letting
+automation take it from there.
+
+Recently, we added support to the operator for [using Flux
+sources](https://www.pulumi.com/docs/guides/continuous-delivery/pulumi-kubernetes-operator/#using-a-flux-source). This
+is a great addition to the operator, but it's not the only way Flux and Pulumi can work together.
+
+Below I'm going to talk about how Flux and Pulumi can be combined, and the superpowers each grants
+to the other.
+
+## Adding OCI support and supply chain security to Pulumi
+
+The support for Flux sources in the Pulumi operator gives you more ways to make your Pulumi programs
+available to the operator. Notably, you can now package your program as an OCI image and push it to
+an image registry, before using it with the operator. This might be appealing if you have deployment
+pipelines to generate Kubernetes YAML (e.g., from [Cue](https://cuelang.org/)) or other data, before
+using it in a program. It's also convenient when you're on a platform like AWS, GCP or Azure,
+because these have OCI registries with useful operational features like caching, security scanning,
+and so on.
+
+But I think an even better reason for using Flux is that it can verify your sources. If you use a
+Flux source in a Pulumi stack, you can take better secure your supply chain. When you're using an
```

Comment:
```suggestion
Flux source in a Pulumi stack, you can take better care of your supply chain. When you're using an
```
or
```suggestion
Flux source in a Pulumi stack, you can secure your supply chain. When you're using an
```

Reply:
I think we just want to remove the word "take" here. In context, Flux can better secure your supply chain by verifying cosign signatures.

```suggestion
Flux source in a Pulumi stack, you can better secure your supply chain. When you're using an
```

Reply:
Agreed and agreed, thanks both.